source/agora/crypto/ECC.d

```diff
@@ -173,13 +173,29 @@ public struct Scalar
         this.data = data;
     }
 
-    /// Construct a scalar from its string representation or a `ubyte[]`
+    /// Construct from its string representation or a fixed length array
     public this (T) (T param)
-        if (is(typeof(this.data = typeof(this.data)(param))))
     {
         this.data = typeof(this.data)(param);
     }
 
+    /// Construct from a dynamic array of the correct length
+    public this (ubyte[data.sizeof] param) inout pure
+    {
+        this.data = typeof(this.data)(cast(ubyte[data.sizeof])(param));
+    }
```

Comment:
Why is this needed ? The previous template constraint should have covered this. And the `cast` should not be necessary, since `param` is already of type `ubyte[data.sizeof]`.

Reply:
The unit test I added shows that the original constructor did not work for all cases. Also it was unclear how the constraint worked. If you prefer we don't include this last commit then can you show me how to get the unit test passing without it please.

Reply:
The problem is:
```
        const from_ubytes = Scalar([57, 34, 14, 84, 18, 175, 101, 64, 121, 181, 212, 78, 23, 148, 180, 7, 9, 105, 237, 155, 78, 161, 191, 27, 97, 130, 209, 44, 202, 245, 208, 13]);
```

The compiler tells you what's wrong:
```
source/agora/crypto/ECC.d(35,35): Error: none of the overloads of this are callable using argument types (int[]), candidates are:
source/agora/crypto/ECC.d(206,13):        agora.crypto.ECC.Scalar.this(BitBlob!32LU data)
source/agora/crypto/ECC.d(219,12):        agora.crypto.ECC.Scalar.this(BitBlob!64LU param)
source/agora/crypto/ECC.d(212,12):        __ctor(T)(T param)
  with T = int[]
  must satisfy the following constraint:
       is(typeof(this.data = typeof(this.data)(param)))
```

That is, passing a literal such as `[1, 2, 3]` infers it as `int[]` not `ubyte[]`. However, with an explicit type, the compiler will try to infer a `ubyte[]`. Now while the explicit overload makes things better, I don't really see the use-case for literals except in unittests.

Reply:
Well in the WellKnownKeys file we have thousands of use cases.

Reply:
I pushed an update with the cast removed as it was not required. I would prefer to keep this extra constructor if you do not object as it will make it easier with the WellKnownKeys file generated from the vanity program.